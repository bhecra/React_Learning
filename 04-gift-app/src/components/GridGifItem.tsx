import { IGiftImage } from '../interfaces/gif.interface';

export const GridGifItem = ({ title, url, id }: IGiftImage) => {
  return (
    <div className='card' key={id}>
      <h2>{title}</h2>
      <img src={url} alt='' />
    </div>
  );
};
