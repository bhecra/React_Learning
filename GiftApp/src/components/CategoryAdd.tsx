import React, { FormEvent, Dispatch, SetStateAction, useState } from 'react';
import PropTypes from 'prop-types';
type Dispatcher<S> = Dispatch<SetStateAction<S>>;

interface IProps {
  setCategories: Dispatcher<string[]>;
}

export const CategoryAdd = (props: IProps) => {
  const setCategories: Dispatcher<string[]> = props.setCategories;
  const [inputValue, setInputValue] = useState('');

  const handleChange = (e: any) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    if (inputValue.trim()) {
      setCategories((item) => [...item, inputValue]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <h2>{inputValue}</h2>
      <input type='text' value={inputValue} onChange={handleChange} />
    </form>
  );
};

CategoryAdd.propTypes = {
  setCategories: PropTypes.oneOf<Dispatcher<string[]>>([]).isRequired,
};
