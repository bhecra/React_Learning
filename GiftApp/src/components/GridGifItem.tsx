import { IGiftImage } from '../interfaces/gif.interface';

interface IGridGiftItemProps {
  img: IGiftImage;
}
export const GridGifItem = ({ title, url, id }: IGiftImage) => {
  return (
    <div className='card' key={id}>
      <h2>{title}</h2>
      <img src={url} alt='' />
    </div>
  );
};
