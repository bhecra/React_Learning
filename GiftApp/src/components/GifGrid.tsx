import React, { useEffect, useState } from 'react';
import { IGiftImage } from '../interfaces/gif.interface';
import { GridGifItem } from './GridGifItem';

interface IGifProps {
  category: string;
}

export const GifGrid = ({ category }: IGifProps) => {
  const [images, setImages] = useState([]);
  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(
      category
    )}&api_key=NTzkwVI55MDAK0AlZdzb2AVUmORkYf3z&limit=10`;
    const response = await fetch(url);
    const { data } = await response.json();

    const gifs = data.map((img: any) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        {images.map((img: any) => {
          return <GridGifItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};
