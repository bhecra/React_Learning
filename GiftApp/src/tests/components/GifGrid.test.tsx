import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';

describe('debe mostrar correctamente el componente', () => {
  test('debe mostrar correctamente', () => {
    let wraper = shallow(<GifGrid category={''} />);

    expect(wraper).toMatchSnapshot();
  });
});
