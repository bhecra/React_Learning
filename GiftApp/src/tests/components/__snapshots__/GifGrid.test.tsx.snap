// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe mostrar correctamente el componente debe mostrar correctamente 1`] = `
<Fragment>
  <h3 />
  Cargando...
  <div
    className="card-grid animate__animated animate__fadeInRight "
  />
</Fragment>
`;
